### PR TITLE
Restyle the color picker a touch

### PR DESCRIPTION
**Pull request by j-f1**
I started out trying to remove all uses of flexbox from the codebase (see branch name) but then I realized the color picker could use some improvement.

Screenshot_2020-03-11 23 12 29

Note: there’s a bug at least in safari where the input loses its `:focus` style when the window is inactive but the `#` icon’s `:focus-within` is still active.

### CODE REVIEW

**Comment by dwelle** [comment]
What's wrong with flex? Grid is IMO unnecessarily complex for simple cases where it's not needed.

**Comment by j-f1** [comment]
I agree, but I think it’s great for things that are actually… grids, like the color picker.

**Comment by dwelle** [comment]
Btw, in case of the color picker, migrating to grid actually made not centered correctly:

before:

image

after:

image

Btw, why remove the color input from the picker? This way when you want to pick a base color and tweak a bit (change hue, reduce opacity...), you need go exit picker first.

**Comment by j-f1** [comment]
Better?

Screenshot_2020-03-12 14 22 12

**Comment by dwelle** [comment]
I think. Btw, there's this weird thing, where the colors' grid squares aren't completely uniform:

image

Not sure if fixed in you latest revision.

(I'm just playing devil's advocate, showing that grid in itself isn't a silver bullet.)

File: src/components/ColorPicker.css [hunk 1]
```diff
@@ -81,33 +81,68 @@
...
   outline: none;
   height: 1.75em;
   box-shadow: #dee2e6 0px 0px 0px 1px inset;
-  box-sizing: content-box;
   border-radius: 0px 4px 4px 0px;
   float: left;
   padding: 1px;
   padding-left: 0.5em;
   appearance: none;
+  font-family: Cascadia, monospace;
```
Comment (lipis) [inline src/components/ColorPicker.css]: At one point I also made that change, but @vjeux convinced me (and I agree) that we should not use Cascadia for "consistency" here..

File: src/components/ColorPicker.css [hunk 2]
```diff
@@ -81,21 +81,42 @@
...
+  height: 100%;
+  position: absolute;
+  top: 0;
+}
+.color-input-container:focus-within .color-picker-hash::before {
+  background: #dee2e6;
+  right: -1px;
+}
+.color-input-container:focus-within .color-picker-hash::after {
+  background: white;
```
Comment (lipis) [inline src/components/ColorPicker.css]:
```suggestion
  background: #fff;
```

**Comment by lipis** [comment]
Shall we get that in?! @vjeux?

**Comment by vjeux** [comment]
Yeah, whenever it’s ready to ship :)

**Pull request merged**